test/jdk/java/lang/RuntimeTests/exec/ExecCommand.java

```diff
@@ -68,163 +84,187 @@ public static String unquote(String str)
+     * and the expected exception for each mode.
+     * Modes:
+     * 0) no SM and default allowAmbiguousCommands; equivalent to true
+     * 1) no SM and allowAmbiguousCommand is empty; equivalent to true
+     * 2) no SM and allowAmbiguousCommands = false
+     * 3) SM and default allowAmbiguousCommands is empty; equivalent to false
+     */
+    @DataProvider(name = "TEST_RTE_ARGS")
+    Object[][] TEST_RTE_ARGS() {
+        return new Object[][]{
+                {"cmd /C dir > dirOut.txt",
+                        "dirOut.txt",
+                        Arrays.asList(null,
+                                null,
+                                FileNotFoundException.class,
+                                FileNotFoundException.class)
+                },
+                {"cmd /C dir > \".\\Program Files\\dirOut.txt\"",
+                        "./Program Files/dirOut.txt",
+                        Arrays.asList(null,
+                                null,
+                                FileNotFoundException.class,
+                                FileNotFoundException.class)
+                },
+                {".\\Program Files\\do.cmd",
+                        null,
+                        Arrays.asList(null,
+                                null,
+                                IOException.class,
+                                IOException.class)
+                },
+                {"\".\\Program Files\\doNot.cmd\" arg",
+                        null,
+                        Arrays.asList(null,
+                                null,
+                                null,
+                                AccessControlException.class)
+                },
+                {"\".\\Program Files\\do.cmd\" arg",
+                        null,
+                        // AccessControlException
+                        Arrays.asList(null, null, null, null)
+                },
+                {"\".\\Program.cmd\" arg",
+                        null,
+                        Arrays.asList(null, null, null, null)
+                },
+                {".\\Program.cmd arg",
+                        null,
+                        Arrays.asList(null, null, null, null)
+                },
+        };
+    }
+
+    /**
+     * Test each command with no SM and default allowAmbiguousCommands.
+     * @param command a command
+     * @param perModeExpected an expected Exception class or null
+     */
+    @Test(dataProvider = "TEST_RTE_ARGS")
+    void testCommandAmbiguousUnset(String command, String testFile, List<Class<Exception>> perModeExpected) {
+        // the JDK_LANG_PROCESS_ALLOW_AMBIGUOUS_COMMANDS is undefined
+        // "true" by default with the legacy verification procedure
+        Properties props = System.getProperties();
+        props.remove(JDK_LANG_PROCESS_ALLOW_AMBIGUOUS_COMMANDS);
+        System.setSecurityManager(null);
+
+        testCommandMode(command, "Ambiguous Unset", testFile, perModeExpected.get(0));
```

Comment:
Any thought to creating a constant for the index value for perModeExpected.get(XX) for extra clarity?